rpcs3/rpcs3qt/rsx_debugger.cpp

```diff
@@ -539,84 +642,561 @@ void rsx_debugger::GetMemory() const
+
+	u32 texture_addr = m_cur_texture;
+	u32 tex_fmt_raw = 0;
+	u32 tex_fmt = 0;
+
+	u32 width = 1280;
+	u32 height = 720;
+	u32 pitch = 0;
+
+	if (texture_addr < 16)
+	{
+		if (textures[texture_addr].enabled())
+		{
+			width = textures[texture_addr].width();
+			height = textures[texture_addr].height();
+			pitch = textures[texture_addr].pitch();
+			tex_fmt_raw = textures[texture_addr].format();
+			tex_fmt = tex_fmt_raw & ~(CELL_GCM_TEXTURE_LN | CELL_GCM_TEXTURE_UN); // TOOD: Support these flags
+			texture_addr = rsx::get_address(textures[texture_addr].offset(), textures[texture_addr].location(), 1);
+		}
+	}
+
+	QString text;
+
+	for (u32 i = 0; i < textures.size(); i++)
+	{
+		// Technically it can also check if used by shader but idk 
+		if (textures[i].enabled() && textures[i].width())
+		{
+			if (!text.isEmpty())
+			{
+				text += ", ";
+			}
 
-	//const u32 width  = render->textures[m_cur_texture].width();
-	//const u32 height = render->textures[m_cur_texture].height();
-	//unsigned char* buffer = (unsigned char*)malloc(width * height * 3);
-	//std::memcpy(buffer, vm::base(TexBuffer_addr), width * height * 3);
+			text += QStringLiteral("%0").arg(i);
+		}
+	}
+
+	m_enabled_textures_label->setText(enabled_textures_text + text);
+
+	if (m_texture_format_override)
+	{
+		tex_fmt = m_texture_format_override;
+	}
+
+	u32 bytes_per_block = 4;
+	u32 pixels_per_block = 1;
+	QImage::Format format = QImage::Format_RGBA8888;
+	bool bswap = true;
+
+	// Naturally only handful of common formats are implemented at the moment
+
+	switch (tex_fmt)
+	{
+	case CELL_GCM_TEXTURE_A8R8G8B8:
+	{
+		format = QImage::Format_RGBA8888;
+		break;
+	}
+	case CELL_GCM_TEXTURE_B8:
+	{
+		format = QImage::Format_Grayscale8;
+		bytes_per_block = 1;
+		break;
+	}
+	case CELL_GCM_TEXTURE_COMPRESSED_DXT1:
+	{
+		bytes_per_block = 8;
+		pixels_per_block = 16;
+		break;
+	}
+	default:
+	{
+		break;
+	}
+	}
+
+	if (!pitch)
+	{
+		pitch = width * bytes_per_block / pixels_per_block;
+	}
+
+	// PS3 buffer size (for memory validation)
+	const u32 src_mem_size = pitch * (height - 1) + width * bytes_per_block / pixels_per_block;
+
+	if (!src_mem_size || !height || !vm::check_addr(texture_addr, vm::page_readable, src_mem_size))
+	{
+		m_buffer_tex->showImage(QImage());
+		return;
+	}
+
+	named_thread("RSX Buffer Touch", [&]()
+	{
+		// Must touch every page
+		for (u32 i = texture_addr & -4096; i < texture_addr + src_mem_size; i += 4096)
+		{
+			static_cast<void>(vm::_ref<atomic_t<u8>>(i).load());
+		}
+	});
+
+	const auto rsx_buffer = vm::get_super_ptr<const u8>(texture_addr);
+	m_buffer_tex->cache.resize(std::max<usz>(m_buffer_tex->cache.size(), width * height * 16));
+
+	const auto buffer = m_buffer_tex->cache.data();
+	if (pixels_per_block != 1 || pitch == width * bytes_per_block / pixels_per_block)
+	{
+		std::memcpy(buffer, rsx_buffer, src_mem_size);
+	}
+	else
+	{
+		for (u32 i = 0; i < src_mem_size; i += (pitch ? pitch : src_mem_size))
```

Comment:
pitch cannot be 0 here. otherwise src_mem_size would be 0.
anyway, even if you wanna keep it as sanity check, it would still be division by 0 in the calculation below and better just added to the if case above